doc/cprover-manual/properties.md

```diff
@@ -0,0 +1,295 @@
+## Automatically Generating Properties
+
+### What is a "Property"?
+
+We have mentioned *properties* several times so far, but we never
+explained *what* kind of properties CBMC can verify. We cover
+this topic in more detail in this section.
+
+CBMC uses
+[assertions](http://en.wikipedia.org/wiki/Assertion_(computing)) to
+specify program properties. Assertions are properties of the state of
+the program when the program reaches a particular program location.
+Assertions are often written by the programmer by means of the `assert`
+macro.
+
+In addition to the assertions written by the programmer, assertions for
+specific properties can also be generated automatically by CBMC, often
+relieving the programmer from writing "obvious" assertions.
+
+CBMC come with an assertion generator, which performs a conservative [static
+analysis](http://en.wikipedia.org/wiki/Static_code_analysis) to determine
+program locations that potentially contain a bug.  Due to the imprecision of
+the static analysis, it is important to emphasize that these generated
+assertions are only *potential* bugs, and that the Model Checker first needs
+to confirm that they are indeed genuine bugs.
+
+The assertion generator can generate assertions for the verification of
+the following properties:
+
+-   **Buffer overflows.** For each array access, check whether the upper
+    and lower bounds are violated.
+-   **Pointer safety.** Search for `NULL`-pointer dereferences or
+    dereferences of other invalid pointers.
+
+-   **Division by zero.** Check whether there is a division by zero in
+    the program.
+
+-   **Not-a-Number.** Check whether floating-point computation may
+    result in [NaNs](http://en.wikipedia.org/wiki/NaN).
+
+-   **Unitialized local.** Check whether the program uses an
+    uninitialized local variable.
+
+-   **Data race.** Check whether a concurrent program accesses a shared
+    variable at the same time in two threads.
```

Comment:
I'm not sure whether we should mention those last two in here, they aren't particularly well maintained or tested.